### PR TITLE
Move more code from the Installation Source spoke

### PR DESCRIPTION
**Pull request by poncovka**
Move some of the methods to new functions and write some tests for them.

### CODE REVIEW

File: tests/unit_tests/pyanaconda_tests/ui/test_ui_source_spoke.py [hunk 1]
```diff
@@ -135,3 +135,18 @@ def test_validate_repo_url(self):
 
         msg = "Repository my_repo required remote directory"
         assert validate_repo_url(PROTOCOL_NFS, "server:", repo_name="my_repo") == msg
+
+    def test_remove_url_prefix(self):
+        """Test the remove_url_prefix function."""
+        # Supported protocols
+        assert remove_url_prefix(PROTOCOL_HTTP, "http://path") == "path"
+        assert remove_url_prefix(PROTOCOL_HTTPS, "https://path") == "path"
+        assert remove_url_prefix(PROTOCOL_FTP, "ftp://path") == "path"
```
Comment (VladimirSlavik) [inline tests/unit_tests/pyanaconda_tests/ui/test_ui_source_spoke.py:144]: I think you also need mismatch between protocol and url, eg:
```
assert remove_url_prefix(PROTOCOL_HTTP, "https://path") == "https://path"
```

File: pyanaconda/ui/gui/spokes/lib/installation_source_helpers.py [hunk 2]
```diff
@@ -231,6 +231,32 @@ def validate_repo_url(protocol, path, repo_name=None):
     return InputCheck.CHECK_OK
 
 
+def remove_url_prefix(protocol, url):
+    """Remove the prefix of the given URL.
+
+    If there is a protocol in the URL, and the protocol matches the
+    combo box, just remove it. This makes it more convenient to paste
+    in URLs. It'll probably freak out people who are typing out http://
+    in the box themselves, but why would you do that?  Don't do that.
```
Comment (VladimirSlavik) [inline pyanaconda/ui/gui/spokes/lib/installation_source_helpers.py:240]: I'm not sure if we want to duplicate that bit :)

File: pyanaconda/ui/gui/spokes/installation_source.py [hunk 3]
```diff
@@ -811,93 +810,36 @@ def _get_selected_partition(self):
             return store[selected][0]
 
     # Input checks
-
-    # This method is shared by the checks on urlEntry and repoUrlEntry
-    def _check_url(self, inputcheck, combo):
+    def _check_url_entry(self, inputcheck):
```
Comment (VladimirSlavik) [inline pyanaconda/ui/gui/spokes/installation_source.py:812]: It might be useful to extend the commit message with the part where `_check_url` was split into two others which both use the new helper. It wasn't exactly clear from the diff...

File: pyanaconda/ui/gui/spokes/lib/installation_source_helpers.py [hunk 4]
```diff
@@ -163,6 +163,74 @@ def check_duplicate_repo_names(repo_names):
...
+    :param repo_name: a name of the repository or None
+    :return: an error message or None
+    """
+    # If this is HTTP/HTTPS/FTP, use the URL_PARSE regex
+    if protocol in (PROTOCOL_HTTP, PROTOCOL_HTTPS, PROTOCOL_FTP):
+        if not path:
+            if repo_name:
+                return _("Repository %s has empty url") % repo_name
+            else:
+                return _("URL is empty")
```
Comment (VladimirSlavik) [inline pyanaconda/ui/gui/spokes/lib/installation_source_helpers.py:180]: I think it could be better to also split the helper into two for repo and non-repo, if you're already refactoring. Having `if repo_name` for each message is weird.

**Review by VladimirSlavik — APPROVED** [review]
Thank you, mostly LGTM, I have only suggestions.

File: pyanaconda/ui/gui/spokes/lib/installation_source_helpers.py (hunk 4 above)
Comment (jkonecny12) [reply pyanaconda/ui/gui/spokes/lib/installation_source_helpers.py:180]: I would even split based on the top-level `if-elif-else` it's pretty hard to follow this way.

**Comment by poncovka** [comment]
I dropped commits that required changes, so all the remaining ones should be ok.
